### PR TITLE
[MM-40214] Implement App Bar icon

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR makes it so the plugin registers its own component to the App Bar. The core webapp makes it so the component is only mounted if the `AppBarEnabled` feature flag is on. This logic is only in the core webapp, and not in the plugin's source.

The plugin registers its usual channel header component, while simultaneously register the App Bar component. This is so the webapp can show one or the other when the App Bar feature flag is enabled or disabled. The plugin itself then doesn't need to watch this feature flag, and the webapp can manage the effects of the flag.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-40214

### CODE REVIEW

File: webapp/src/actions.js [hunk 1]
```diff
@@ -53,29 +53,6 @@ export function updateRhsState(rhsState) {
...
-    return basePath + '/plugins/' + pluginId;
-};
-
-export const telemetry = (event, properties) => async (dispatch, getState) => {
-    await fetch(getPluginServerRoute(getState()) + '/telemetry', Client4.getOptions({
-        method: 'post',
-        body: JSON.stringify({event, properties}),
-    }));
-};
-
```
Comment (hanzei) [inline webapp/src/actions.js:78]: Is it worth removing the unused method? We might want to add telemetry at some point.

File: webapp/src/actions.js [hunk 2]
```diff
@@ -53,29 +53,6 @@ export function updateRhsState(rhsState) {
     };
 }
 
-// TODO: Move this into mattermost-redux or mattermost-webapp.
```
Comment (hanzei) [inline webapp/src/actions.js:56]: Is this still a TODO?

**Review by hanzei — COMMENTED** [review]
An ico, awesome :tada: Should the new icon also be used for the Marketplace as a plugin icon?

**Review by larkox — APPROVED** [review]

**Comment by mickmister** [comment]
> An icon, awesome 🎉 Should the new icon also be used for the Marketplace as a plugin icon?

@hanzei It can't be used for the marketplace because it's a png file, and the marketplace requires svg files.

cc @matthewbirtch as this is a good point that some plugins are missing marketplace icons, and we are not able to reuse the same assets provided for the App Bar. I've created a ticket to add more icons to the marketplace https://mattermost.atlassian.net/browse/MM-41024

**Comment by hanzei** [comment]
> @hanzei It can't be used for the marketplace because it's a png file, and the marketplace requires svg files.

Is it reasonable to convert the icon to an `svg`?

**Comment by matthewbirtch** [comment]
> > An icon, awesome 🎉 Should the new icon also be used for the Marketplace as a plugin icon?
> 
> @hanzei It can't be used for the marketplace because it's a png file, and the marketplace requires svg files.
> 
> cc @matthewbirtch as this is a good point that some plugins are missing marketplace icons, and we are not able to reuse the same assets provided for the App Bar. I've created a ticket to add more icons to the marketplace https://mattermost.atlassian.net/browse/MM-41024

Wouldn't it be better if plugin developers just had to worry about one file for their plugin's icon that worked for in-product and the marketplace?

**Comment by mickmister** [comment]
@hanzei @matthewbirtch 

> Wouldn't it be better if plugin developers just had to worry about one file for their plugin's icon that worked for in-product and the marketplace?

Yes that certainly would be better 🙂  There was a start of a discussion of why we went with requiring SVG icons for the marketplace, but there wasn't a conclusion to the conversation https://github.com/mattermost/mattermost-marketplace/pull/35#issuecomment-981800259. It's possible we continued that discussion somewhere else but I'm not sure if we did.

@matthewbirtch I believe we went with PNGs for the App Bar icons to make it easier to match the brand of the external apps, as well as making it easier for developers to create icons, though they still have the requirement of SVGs for the marketplace, so we're not completely removing that obstacle.

**Comment by matthewbirtch** [comment]
> @matthewbirtch I believe we went with PNGs for the App Bar icons to make it easier to match the brand of the external apps, as well as making it easier for developers to create icons, though they still have the requirement of SVGs for the marketplace, so we're not completely removing that obstacle.

Yes, PNGs would certainly be easier for developers. What's involved in allowing PNGs in the marketplace so that developers only need to worry about updating 1 file? I don't think we concluded that conversation

**Comment by mickmister** [comment]
> What's involved in allowing PNGs in the marketplace so that developers only need to worry about updating 1 file? I don't think we concluded that conversation

@matthewbirtch I know there is code in `mattermost-server` and `mattermost-marketplace` that relies on the image being in svg format, because the image is converted into a data URL, with the data type being hardcoded as `svg`:

https://github.com/mattermost/mattermost-marketplace/blob/f828534e8783306038865c496580f01d8fe233a7/cmd/generator/main.go#L448

```go
func getIconDataFromTarFile(file []byte, path string) (string, error) {
	iconData, err := getFromTarFile(tar.NewReader(bytes.NewReader(file)), path)
	if err != nil {
		return "", errors.Wrapf(err, "failed to read icon data from plugin bundle for path %s", path)
	}

        // svg format is hardcoded here
	return fmt.Sprintf("data:image/svg+xml;base64,%s", base64.StdEncoding.EncodeToString(iconData)), nil
}
```

This requirement could in theory be removed, but it's important to discuss why it was made required. I think @hanzei may have some context because the change in the marketplace code is in his PR here https://github.com/mattermost/mattermost-marketplace/pull/35

> so that developers only need to worry about updating 1 file?

Is allowing PNG the best way to solve this? I'm asking because I legitimately don't know the answer.

**Comment by matthewbirtch** [comment]
> Is allowing PNG the best way to solve this? I'm asking because I legitimately don't know the answer.

I suppose that's up for debate. However, creating PNGs is certainly much easier than creating SVGs, depending on what software you might have. I guess that's one of the main reasons I've pursued this route. 

If we need to move forward with SVGs being the definitive icon format, I'm not against that, but then we should enforce that across the board for all locations of the icon (bot avatars, apps bar, marketplace icon, etc)

**Comment by hanzei** [comment]
Thanks for raising these questions @matthewbirtch and @mickmister. I've queued a agenda item to discuss the plans moving forward: https://community-daily.mattermost.com/core/pl/god7fpu4sjfrmntugzun7ri6rc

File: webapp/src/actions.js (hunk 2 above)
Comment (hanzei) [reply webapp/src/actions.js:56]: 1/5 to keep it

Comment (mickmister) [reply webapp/src/actions.js:56]: Done

**Review by hanzei — APPROVED** [review]

**Comment by mickmister** [comment]
@DHaussermann is this good to merge?

**Pull request merged**